RIOT-OS/RIOT
# testing: add periph I2C robot tests for HIL

### Contribution description

This adds a new test suite for the I2C peripheral drivers based on the RobotFramework and PHiLIP to allow for hardware-in-the-loop (HIL) testing.

### Testing procedure

To run this you need a board (device under test) hooked up to a PHILIP test board, see https://github.com/riot-appstore/PHiLIP . Then run `BOARD=<fill in> make -C tests/periph_i2c flash robot-test`.

### Issues/PRs references

depends on #10095
